crates/gritcache/src/new_cache.rs

```diff
@@ -39,8 +39,8 @@ impl ThreadedCache {
         };
 
         let (sender, receiver) = mpsc::channel::<HashKey>();
+        let mut writer = Self::new_writer(&mismatches_path)?;
```

Comment:
@morgante let me know if there's a reason not to do this i'm missing - seems more straightforward to bail during creation than to do so from the writer thread

Reply:
This looks fine - if the compiler is happy, I'm happy.